pkg/handlers/handlers.go

```diff
@@ -27,10 +28,18 @@ func HandleFile(file io.Reader, chunkSkel *sources.Chunk, chunksChan chan (*sour
 			continue
 		}
 		handlerChan := handler.FromFile(file)
-		for data := range handlerChan {
-			chunk := *chunkSkel
-			chunk.Data = data
-			chunksChan <- &chunk
+		for {
+			select {
+			case data := <-handlerChan:
+				chunk := *chunkSkel
+				chunk.Data = data
+				chunksChan <- &chunk
+			case <-ctx.Done():
+				break
+			}
```

Comment:
**Suggestion:** `break` here actually breaks out of the `select`, not the outter `for` :/

Maybe `return true` or use a labelled `for`?

Reply:
Thanks for catching that!

Reply:
It's bitten me before 😅